File: backend/app/schemas/document.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class DocumentUploadResponse(BaseModel):
    document_id: uuid.UUID
    filename: str
    chunks_created: int
    status: str


class DocumentListItem(BaseModel):
    id: uuid.UUID
    filename: str
    source_type: str | None
    category: str | None
    chunks_count: int
    created_at: datetime

File: frontend/src/types/document.ts
export type DocumentListItem = {
  id: string;
  filename: string;
  source_type: string | null;
  category: string | null;
  chunks_count: number;
  created_at: string;
};

export type DocumentUploadResponse = {
  document_id: string;
  filename: string;
  chunks_created: number;
  status: string;
};
